### PR TITLE
#403 - Introduce `Builder` for `Validator` and remove `JwsSignerObject`

### PR DESCRIPTION
**Pull request by duttarnab**
#403 - Introduce `Builder` for `Validator` and remove `JwsSignerObject`
https://github.com/GluuFederation/oxd/issues/403

### CODE REVIEW

File: oxd-server/src/main/java/org/gluu/oxd/server/op/Validator.java [hunk 1]
```diff
@@ -35,53 +37,107 @@
...
+        this.discoveryResponse = builder.discoveryResponse;
+        this.idToken = builder.idToken;
+        this.opClientFactory = builder.opClientFactory;
+        this.keyService = builder.keyService;
+        this.rp = builder.rp;
+        this.jwsSigner = createJwsSigner(idToken, discoveryResponse, keyService, opClientFactory, rp);
+    }
+
+    //Builder Class
+    public static class ValidatorBuilder {
```
Comment (yuriyz) [inline oxd-server/src/main/java/org/gluu/oxd/server/op/Validator.java]: ValidationBuilder -> Builder

File: oxd-server/src/main/java/org/gluu/oxd/server/op/Validator.java [hunk 2]
```diff
@@ -35,53 +37,107 @@
...
+                                OpClientFactory opClientFactory,
+                                PublicOpKeyService keyService,
+                                Rp rp) {
+
+            this.discoveryResponse = discoveryResponse;
+            this.idToken = idToken;
+            this.opClientFactory = opClientFactory;
+            this.keyService = keyService;
+            this.rp = rp;
+        }
```
Comment (yuriyz) [inline oxd-server/src/main/java/org/gluu/oxd/server/op/Validator.java]: There is no sense in builder if there are no setters methods
```
Validator.builder()
    .discoveryResponse(discoveryResponse)
    .idToken(idToken)
    .opClientFactory(opClientFactory)
    .keyService(keyService)
    .rp(rp)
    .build();
```

File: oxd-server/src/main/java/org/gluu/oxd/server/op/Validator.java [hunk 3]
```diff
@@ -35,53 +37,107 @@
...
+
+        public Validator build() {
```
Comment (yuriyz) [inline oxd-server/src/main/java/org/gluu/oxd/server/op/Validator.java]: before building validator we should have preconditions. For example there is not sense in Validator if idToken is null.
`Preconditions.checkNotNull(idToken)`.

File: oxd-server/src/main/java/org/gluu/oxd/server/op/Validator.java (hunk 1 above)
Comment (duttarnab) [reply oxd-server/src/main/java/org/gluu/oxd/server/op/Validator.java]: Done

File: oxd-server/src/main/java/org/gluu/oxd/server/op/Validator.java (hunk 2 above)
Comment (duttarnab) [reply oxd-server/src/main/java/org/gluu/oxd/server/op/Validator.java]: Done

File: oxd-server/src/main/java/org/gluu/oxd/server/op/Validator.java (hunk 3 above)
Comment (duttarnab) [reply oxd-server/src/main/java/org/gluu/oxd/server/op/Validator.java]: Done

**Pull request merged**